wordpress-mobile/WordPress-iOS
# App Settings Cleanup, Part 1

Ref #11925

## Summary
Part 1 of this PR updates some of the UI in App Settings to follow our current color scheme and improves the UX for clearing the Media Cache. This PR does not address the changes to the "Max Image Upload Size" row or the new location option. It is arriving in a separate future PR.

**Changes include:**
- Clear Spotlight Index button uses blue brand color and is left aligned
- Clear Siri Shortcut Suggestions uses blue brand color and is left aligned
- Media Cache Size UI is updated to show a disclosure arrow and navigates to its own ViewController 
- "Other" section header remains intact because more options are listed in that section compared to when the ticket was created

## Screenshots
| Before | After |
| --- | --- |
| ![Simulator Screen Shot - iPhone 12 Pro - 2022-05-04 at 14 20 06](https://user-images.githubusercontent.com/1062444/167921306-c5ee76b8-8add-41a0-a376-77ed74e90ee5.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2022-05-04 at 16 04 50](https://user-images.githubusercontent.com/1062444/167921360-74a9e31e-48ce-42f1-93ef-0a7c48ee1fae.png) |

## To test:
0. Install the app on a device, with iOS 12 or higher. Steps 7 and 8 are features found only in iOS 12 and up.
1. Select "My Site" in the bottom nav baar
2. Select your profile picture
3. Choose "App Settings" in the modal view
4. The "Media Cache Size" row should display the current size of the media cache
5. Select the row. It should navigate to a new ViewController with the "Media Cache Size" title and size row, as well as a row button to clear the cache. Clear the cache and verify the size reports "Empty"
6. Navigate back to App Settings
7. Tap the "Clear Spotlight Index" and wait for a toast message that it was successful
8. Tap the "Clear Siri Shortcut Suggestions" and wait for a toast message that it was successful 

## Regression Notes
1. Potential unintended areas of impact
- App Settings
- WPImmuTableRows contains a new row button style

6. What I did to test those areas of impact (or what existing automated tests I relied on)
Run all unit tests, which includes:
- ImmuTableTest 
- MediaFileManagerTests 

10. What automated tests I added (or what prevented me from doing so)
- None, UI tests don't seem like a valuable use of time

PR submission checklist:
